src/hw/hw_mngr.cpp

```diff
@@ -357,3 +563,264 @@ void HWManager::setFanSpeedDelayed()
         setFanSpeed();
     });
 }
+
+void HWManager::saveSystemFanFeatures()
+{
+    log<level::DEBUG>("HWManager::saveSystemFanFeatures()");
+
+    std::ofstream ofs(sysFanDataFile);
+    json  jsonDoc;
+
+    for (const auto& [fanIndex, fanFeature] : fanFeatures)
+    {
+        json  jsonItem;
+        jsonItem["fanIndex"] = fanIndex;
+        jsonItem["partNumber"] = fanFeature.partNumber;
+        jsonItem["prettyName"] = fanFeature.prettyName;
+        jsonItem["type"] = fanFeature.type;
+        jsonDoc.push_back(jsonItem);
+    }
+
+    ofs << std::setw(4) << jsonDoc << std::endl;
+}
+
+void HWManager::loadSystemFanFeatures()
+{
+    log<level::DEBUG>("HWManager::loadSystemFanFeatures()");
+    fanFeatures.clear();
+
+    std::ifstream dataFile(sysFanDataFile);
+    if (!dataFile.is_open())
+    {
+        log<level::DEBUG>("Cannot to open file",
```

Comment:
`level::DEBUG`?
I guess we should check if file exists before an opening attempt and show this error with `level::ERROR` or even throw an exception.

Reply:
this is not a bug. this is cache file which can be not exist. In this case, we do FAN detection.

Reply:
And what happens if the parent directory doesn't exist or is read-only?

Reply:
If we can't read file, the fanFeatures will be empty - at next time we do FAN detection and then update a file.
if we read file and format is wrong, the fanFeatures will be empty - at next time we do FAN detection and then update a file. 
if we read file and successfully parse a content, we get a not empty fanFeatures. FAN detection will be skipped.